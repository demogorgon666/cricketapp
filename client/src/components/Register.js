import React, { Component } from "react";
import { Link } from "react-router-dom";
import { getUsers, createUsers } from "../actions/Users.js";
import { connect } from "react-redux";
import "./css/Login.css";

export class Register extends Component {
	// constructor(props) {
	// 	super(props);
	// 	this.onRegister = this.onRegister.bind(this);
	// }

	// componentDidMount() {
	//   this.props.getUsers();
	// }

	componentDidMount() {
		// if (localStorage.getItem("token")) {
		//   this.props.history.push("/");
		// }
	}

	state = {
		user_name: "",
		user_email: "",
		user_password: "",
		isadmin: false,
		confirmPassword: "",
		showError: false,
		errorMessage: ""
	};

	OnChange = event => {
		this.setState({ [event.target.name]: event.target.value });
	};

	onRegister = e => {
		e.preventDefault();
		if (
			!this.state.user_name ||
			!this.state.user_email ||
			!this.state.user_password ||
			!this.state.confirmPassword
		) {
			this.setState({
				showError: true,
				errorMessage: "Enter all the fields"
			});
		} else {
			//   e.preventDefault();
			if (this.state.user_password !== this.state.confirmPassword) {
				this.setState({
					showError: true,
					errorMessage: "Password does not match"
				});
			} else {
				e.preventDefault();
				if (this.state.user_name.length < 3) {
					this.setState({
						showError: true,
						errorMessage: "Name length should be greater then 3!"
					});
				} else {
					if (this.state.user_password.length < 5) {
						this.setState({
							showError: true,
							errorMessage:
								"Password length should be greater then 5!"
						});
					} else {
						if (this.state.user_email.length < 5) {
							this.setState({
								showError: true,
								errorMessage: "Please enter valid email!"
							});
						} else {
							this.setState({
								showError: false,
								errorMessage: ""
							});
							let user = {
								user_name: this.state.user_name,
								user_email: this.state.user_email,
								user_password: this.state.user_password,
								isadmin: this.state.isadmin
							};
							this.props.createUsers(user);
							// this.props.history.push("/login");
							this.setState({
								user_name: "",
								user_email: "",
								user_password: "",
								confirmPassword: ""
							});
						}
					}
				}
			}
		}
	};

	render() {
		return (
			<div>
				<form id="msform">
					<fieldset>
						<h1>Register</h1>
						<input
							type="text"
							name="user_name"
							placeholder="Enter Name"
							value={this.state.user_name}
							onChange={this.OnChange}
						/>
						<input
							type="text"
							name="user_email"
							placeholder="Enter Email"
							value={this.state.user_email}
							onChange={this.OnChange}
						/>
						<input
							type="password"
							name="user_password"
							placeholder="Enter Password"
							value={this.state.user_password}
							onChange={this.OnChange}
						/>
						<input
							type="password"
							name="confirmPassword"
							placeholder="Confirm Password"
							value={this.state.confirmPassword}
							onChange={this.OnChange}
						/>
						<div
							className="errorMessage"
							style={{
								color: "#c0392b"
							}}
						>
							{/*dispatch error from node*/}
							{this.props.error ? (
								<>{this.props.error}</>
							) : (
								<span
									className="errorMessage"
									style={{
										color: "#c0392b",
										display: this.state.showError
											? "block"
											: "none"
									}}
								>
									{this.state.errorMessage}
								</span>
							)}
						</div>

						<button
							className="formbutton"
							onChange={this.OnChange}
							onClick={this.onRegister}
						>
							Sign Up
						</button>
						<p>
							Already have an account ?
							<Link to="/login">
								<span
									style={{
										color: "#2980b9",
										fontWeight: "bold"
									}}
								>
									Login
								</span>
							</Link>
						</p>
					</fieldset>
				</form>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	users: state.userReducer.users,
	error: state.userReducer.error
});

export default connect(
	mapStateToProps,
	{ getUsers, createUsers }
)(Register);
