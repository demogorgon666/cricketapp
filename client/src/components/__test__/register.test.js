import React from "react";
import { shallow, mount } from "enzyme";
import { Register } from "../Register";
import store from "../../store";
import { Provider } from "react-redux";
const register = jest.fn();

const wrapper = shallow(<Register register={register} />);

describe("test  Component", () => {
  it("render the component", () => {
    // const wrapper = shallow(<Register register={register} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("there should be a div", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("div").length).toBe(2);
  });
  it("there should be a navbar", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("NavBar").length).toBe(0);
  });
  it("there should be a h1", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("h1").length).toBe(1);
    expect(wrapper.find("h1").text()).toBe("Register");
  });
  it("there should be a input", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("input").length).toBe(4);
  });
  it("there should be a buttons", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("button").length).toBe(1);
  });
  it("there should be a link", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("Link").length).toBe(1);
  });
  it("there should be a form", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("form").length).toBe(1);
  });
  it("there should be a fieldset tag", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("fieldset").length).toBe(1);
  });
  it("there should be a p tag", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("p").length).toBe(1);
    // expect(wrapper.find("p").text()).toBe("Already have an account ?");
  });
  it("there should be a button", () => {
    // const wrapper = mount(<Register register={register} />);
    expect(wrapper.find("button").text()).toBe("Sign Up");
  });
});
