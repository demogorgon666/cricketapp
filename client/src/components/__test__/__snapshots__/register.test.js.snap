// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  Component render the component 1`] = `
<div>
  <Connect(Navbar) />
  <form
    id="msform"
  >
    <fieldset>
      <h1>
        Register
      </h1>
      <input
        name="user_name"
        onChange={[Function]}
        placeholder="Enter Name"
        type="text"
        value=""
      />
      <input
        name="user_email"
        onChange={[Function]}
        placeholder="Enter Email"
        type="text"
        value=""
      />
      <input
        name="user_password"
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value=""
      />
      <input
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm Password"
        type="password"
        value=""
      />
      <div
        className="errorMessage"
        style={
          Object {
            "color": "#c0392b",
          }
        }
      >
        <span
          className="errorMessage"
          style={
            Object {
              "color": "#c0392b",
              "display": "none",
            }
          }
        />
      </div>
      <button
        className="formbutton"
        onChange={[Function]}
        onClick={[Function]}
      >
        Sign Up
      </button>
      <p>
        Already have an account ?
        <Link
          to="/login"
        >
          <span
            style={
              Object {
                "color": "#2980b9",
                "fontWeight": "bold",
              }
            }
          >
            Login
          </span>
        </Link>
      </p>
    </fieldset>
  </form>
</div>
`;
